fault/tester.py

```diff
@@ -33,18 +34,33 @@ def make_target(self, target, **kwargs):
                                         **kwargs)
         raise NotImplementedError(target)
 
+    def make_array_action(self, port, value, action):
+        for i in range(port.N):
+            action(port[i], value[i])
+
     def poke(self, port, value):
         value = make_value(port, value)
-        self.actions.append(actions.Poke(port, value))
```

Comment:
I think this logic should be delegated to the targets -- tester should ideally not be implementing logic to split a `poke(array_port, [...])` into several actions. I would argue these functions should be unchanged, and in the targets we do a:

```python
if isinstance(value, fault.Array):
   ...
```

Because, the magma/coreir simulator may support directly `set_value(array_port, [...])` where as of course for verilator we need to blow it up.

Reply:
Good point, that's actually how I initially implemented it. I'll make this change.